test/unit/api_spec.js

```diff
@@ -798,135 +742,103 @@ describe("api", function () {
         return pdfDoc.getPageLabels();
       });
 
-      Promise.all([promise0, promise1, promise2, promise3])
-        .then(function (pageLabels) {
-          expect(pageLabels[0]).toEqual(["i", "ii", "iii", "1"]);
-          expect(pageLabels[1]).toEqual(["Front Page1"]);
-          expect(pageLabels[2]).toEqual(["1", "2"]);
-          expect(pageLabels[3]).toEqual(["X3"]);
-
-          Promise.all([
-            loadingTask0.destroy(),
-            loadingTask1.destroy(),
-            loadingTask2.destroy(),
-            loadingTask3.destroy(),
-          ]).then(done);
-        })
-        .catch(done.fail);
+      const pageLabels = await Promise.all([
+        promise0,
+        promise1,
+        promise2,
+        promise3,
+      ]);
+      expect(pageLabels[0]).toEqual(["i", "ii", "iii", "1"]);
+      expect(pageLabels[1]).toEqual(["Front Page1"]);
+      expect(pageLabels[2]).toEqual(["1", "2"]);
+      expect(pageLabels[3]).toEqual(["X3"]);
+
+      await Promise.all([
+        loadingTask0.destroy(),
+        loadingTask1.destroy(),
+        loadingTask2.destroy(),
+        loadingTask3.destroy(),
+      ]);
     });
 
-    it("gets default page layout", function (done) {
+    it("gets default page layout", async function () {
       const loadingTask = getDocument(
         buildGetDocumentParams("tracemonkey.pdf")
       );
 
-      loadingTask.promise
-        .then(function (pdfDoc) {
-          return pdfDoc.getPageLayout();
-        })
-        .then(function (mode) {
-          expect(mode).toEqual("");
+      const pageLayout = await loadingTask.promise.then(function (pdfDoc) {
+        return pdfDoc.getPageLayout();
+      });
```

Comment:
Nit: Here, and in *a lot* of similar cases below (where there's only one promise involved), you could just do e.g. the following instead.
```javascript
const pdfDoc = await loadingTask.promise;
const pageLayout = await pdfDoc.getPageLayout();
```

Reply:
Good idea! This increases readability and saves another 30-ish lines in the file.